pkg/dns/serviceresolver.go

```diff
@@ -128,7 +128,7 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 		}
 
 		// no clusterIP and not headless, no DNS
-		if len(svc.Spec.ClusterIP) == 0 {
+		if len(svc.Spec.ClusterIP) == 0 && svc.Spec.Type != kapi.ServiceTypeExternalName {
```

Comment:
You can already set up a proxy at that name that points to any server.  TLS only trusts the name the user gives, not the other way around.  Also, you trust the name and the thing the name points to already.